diff -Naur a/src/compat.ml b/src/compat.ml
--- a/src/compat.ml	1970-01-01 00:00:00.000000000 +0000
+++ b/src/compat.ml	2020-11-24 09:04:43.959367163 +0000
@@ -0,0 +1,7 @@
+if Filename.check_suffix Sys.argv.(1) ".ml" &&
+   Scanf.sscanf Sys.ocaml_version "%d.%d" (fun a b -> (a, b)) < (4, 03) then
+  print_endline "\
+module String = struct
+  include String
+  let capitalize_ascii = capitalize
+end"
diff -Naur a/src/dune b/src/dune
--- a/src/dune	2019-05-26 23:49:42.000000000 +0000
+++ b/src/dune	2020-11-24 09:04:43.959367163 +0000
@@ -13,4 +13,9 @@
  (name cppo_main)
  (package cppo)
  (public_name cppo)
+ (modules :standard \ compat)
+ (preprocess (per_module
+  ((action (progn
+    (run ocaml %{dep:compat.ml} %{input-file})
+    (cat %{input-file}))) cppo_eval)))
  (libraries unix str))
