diff -Naur a/Changes b/Changes
--- a/Changes	2019-05-26 23:49:42.000000000 +0000
+++ b/Changes	2020-11-18 16:48:22.436525214 +0000
@@ -1,3 +1,7 @@
+## v1.6.7 (2020-??-??)
+- [compat] Treat ~ and - the same in semver in order to parse
+           OCaml 4.12.0 pre-release versions.
+
 ## v1.6.6 (2019-05-27)
 - [pkg] port build system to dune from jbuilder.
 - [pkg] upgrade opam metadata to 2.0 format.
diff -Naur a/src/cppo_main.ml b/src/cppo_main.ml
--- a/src/cppo_main.ml	2019-05-26 23:49:42.000000000 +0000
+++ b/src/cppo_main.ml	2020-11-18 16:48:22.436525214 +0000
@@ -18,7 +18,7 @@
 \\([0-9]+\\)\
 \\.\\([0-9]+\\)\
 \\.\\([0-9]+\\)\
-\\(-\\([^+]*\\)\\)?\
+\\([~-]\\([^+]*\\)\\)?\
 \\(\\+\\(.*\\)\\)?\
 \r?$"
 
@@ -108,6 +108,10 @@
             VAR_VERSION_FULL    is the original string
 
           Example: cppo -V OCAML:4.02.1
+
+          Note that cppo recognises both '-' and '~' preceding the pre-release
+          meaning -V OCAML:4.11.0+alpha1 sets OCAML_BUILD to alpha1 but
+          -V OCAML:4.12.0~alpha1 sets OCAML_PRERELEASE to alpha1.
 ";
 
     "-o", Arg.String (fun s -> out_file := Some s),
