diff -Naur a/src/cppo_eval.ml b/src/cppo_eval.ml
--- a/src/cppo_eval.ml	2019-05-26 23:49:42.000000000 +0000
+++ b/src/cppo_eval.ml	2019-11-21 14:29:33.416891946 +0000
@@ -82,11 +82,10 @@
   let buf = Buffer.create (String.length s) in
   trim_and_compact buf s;
   Buffer.contents buf
-
 let trim_compact_and_capitalize_string s =
   let buf = Buffer.create (String.length s) in
   trim_and_compact buf s;
-  String.capitalize_ascii (Buffer.contents buf)
+  String.capitalize (Buffer.contents buf)
 
 let is_ident s =
   let len = String.length s in
