diff -Naur a/_tags b/_tags
--- a/_tags	2017-03-07 14:02:34.000000000 +0000
+++ b/_tags	2022-03-09 19:48:12.868332000 +0000
@@ -1,4 +1,4 @@
-true : bin_annot, safe_string, package(bytes), package(uchar), package(uutf)
+true : bin_annot, safe_string, package(bytes), package(uutf)
 <src> : include
 <test> : include
 <test/jsontrip.*> : package(unix)
diff -Naur a/pkg/META b/pkg/META
--- a/pkg/META	2017-03-07 14:02:34.000000000 +0000
+++ b/pkg/META	2022-03-09 18:51:52.343637213 +0000
@@ -1,6 +1,6 @@
 description = "Non-blocking streaming JSON codec"
 version = "1.0.1"
-requires = "uchar bytes uutf"
+requires = "bytes uutf"
 archive(byte) = "jsonm.cma"
 archive(native) = "jsonm.cmxa"
 plugin(byte) = "jsonm.cma"
