diff -Naur a/Makefile b/Makefile
--- a/Makefile	2017-03-21 14:39:23.000000000 +0000
+++ b/Makefile	2019-11-21 14:29:33.684757960 +0000
@@ -2,6 +2,8 @@
 
 NAME = cudf
 
+EXT_LIB=$(shell ocamlc -config | fgrep ext_lib | sed -e "s/ext_lib: \.//")
+
 ifeq ("$(shell (ocamlc -config 2>/dev/null || ocamlopt -config) | fgrep os_type)","os_type: Win32")
 EXE=.exe
 OCAMLLIBDIR := $(shell cygpath $(OCAMLLIBDIR))
@@ -81,9 +83,9 @@
 	otags $^
 
 INSTALL_STUFF = META
-INSTALL_STUFF += $(wildcard _build/*.cma _build/*.cmxa _build/cudf.a)
+INSTALL_STUFF += $(wildcard _build/*.cma _build/*.cmxa _build/cudf.$(EXT_LIB))
 INSTALL_STUFF += $(wildcard _build/cudf_*.cmi) $(wildcard _build/*.mli)
-INSTALL_STUFF += $(wildcard _build/cudf_*.cmx _build/cudf_*.o _build/cudf_*.a)
+INSTALL_STUFF += $(wildcard _build/cudf_*.cmx _build/cudf_*.o _build/cudf_*.$(EXT_LIB))
 INSTALL_STUFF += $(wildcard _build/cudf.o _build/cudf.cmx _build/cudf.cmi)
 
 install:
