diff -Naur a/Makefile.config.in b/Makefile.config.in
--- a/Makefile.config.in	2016-07-20 11:45:05.000000000 +0000
+++ b/Makefile.config.in	2019-11-21 14:29:34.508346003 +0000
@@ -12,14 +12,14 @@
 BINDIR = @prefix@@bindir@
 # if prefix is /usr/local (default), then we use ocamlfind, 
 # otherwise use build the destdir using the given prefix
-ifeq (@prefix@,/usr/local)
-  ifeq ("@OCAML_OS_TYPE@","Win32")
-    LIBDIR=$(shell ocamlfind printconf destdir | cygpath -f - -m)
-  else
+ifeq ("@OCAML_OS_TYPE@","Win32")
+  LIBDIR=$(shell ocamlfind printconf destdir | cygpath -f - -m)
+else
+  ifeq (@prefix@,/usr/local)
     LIBDIR = $(shell ocamlfind printconf destdir)
+  else
+    LIBDIR = @prefix@@libdir@/ocaml/$(shell ocamlc -version)
   endif
-else
-  LIBDIR = @prefix@@libdir@/ocaml/$(shell ocamlc -version)
 endif
 
 # if DESTDIR is specified, we ignore the prefix and we use the 
