diff -Naur a/Makefile b/Makefile
--- a/Makefile	2016-07-20 11:45:05.000000000 +0000
+++ b/Makefile	2019-11-21 14:29:34.552324006 +0000
@@ -222,7 +222,7 @@
 INSTALL_STUFF_ = META
 INSTALL_STUFF_ += $(wildcard _build/doselibs/*.cma _build/doselibs/*.cmi)
 INSTALL_STUFF_ += $(wildcard _build/doselibs/*.cmxa _build/doselibs/*.cmxs)
-INSTALL_STUFF_ += $(wildcard _build/doselibs/*.a)
+INSTALL_STUFF_ += $(wildcard _build/doselibs/*.$(EXT_LIB))
 #INSTALL_STUFF_ += $(wildcard _build/*/*.mli)
 INSTALL_STUFF_ += $(wildcard _build/rpm/*.so)
 
diff -Naur a/Makefile.config.in b/Makefile.config.in
--- a/Makefile.config.in	2019-11-21 14:29:34.508346003 +0000
+++ b/Makefile.config.in	2019-11-21 14:29:34.552324006 +0000
@@ -134,12 +134,14 @@
 endif
 
 ifeq ("@OCAMLEXT@","native")
-  SUFFIX += cmxa cmxs a
+  SUFFIX += cmxa cmxs @EXT_LIB@
   OPTLIBS += $(BYTELIBS:%.cma=%.cmxa) 
   CMXSLIBS += $(BYTELIBS:%.cma=%.cmxs)
-  ALIBS = $(BYTELIBS:%.cma=%.a)
+  ALIBS = $(BYTELIBS:%.cma=%.@EXT_LIB@)
 endif
 
+EXT_LIB=@EXT_LIB@
+
 ifeq ("@HAS_XML@","yes")
 ifeq ("@HAS_CURL@","yes")
   EXPERIMENTAL += experimental/dudftocudf/deb-dudftocudf.@OCAMLEXT@
diff -Naur a/configure b/configure
--- a/configure	2016-07-20 11:45:05.000000000 +0000
+++ b/configure	2019-11-21 14:29:34.552324006 +0000
@@ -648,6 +648,7 @@
 CONFIG_ZIP
 CPPOFLAGS
 LN
+EXT_LIB
 OCAML_SYSTEM
 EGREP
 GREP
@@ -5317,6 +5318,8 @@
 
 OCAML_SYSTEM="$($OCAMLBESTCC -config | fgrep system | sed -e "s/system: \(.*\)/\1/")"
 
+EXT_LIB="$($OCAMLBESTCC -config | fgrep ext_lib | sed -e "s/ext_lib: \.//")"
+
 
 if test "${OCAML_OS_TYPE}" = "Win32"; then :
 
diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2016-07-20 11:45:05.000000000 +0000
+++ b/configure.ac	2019-11-21 14:29:34.552324006 +0000
@@ -256,6 +256,8 @@
 
 OCAML_SYSTEM="$($OCAMLBESTCC -config | fgrep system | sed -e "s/system: \(.*\)/\1/")"
 AC_SUBST(OCAML_SYSTEM)
+EXT_LIB="$($OCAMLBESTCC -config | fgrep ext_lib | sed -e "s/ext_lib: \.//")"
+AC_SUBST(EXT_LIB)
 
 AS_IF([test "${OCAML_OS_TYPE}" = "Win32"],[
   AC_MSG_CHECKING([for a workable solution for ln -s])
