diff -Naur a/Makefile b/Makefile
--- a/Makefile	2019-11-21 14:29:34.580310007 +0000
+++ b/Makefile	2019-11-21 14:29:34.620290009 +0000
@@ -1,5 +1,6 @@
 #this is a forward reference to the target all below
-all: all
+.PHONY: default
+default: all
 
 #SHELL=/bin/bash
 include Makefile.config
@@ -15,8 +16,8 @@
 #OBFLAGS := $(OBFLAGS) -classic-display
 
 addnotrpm:
-	@if [ ! -s doseparseNoRpm ]; then $(LN) doseparse doseparseNoRpm ; fi
-	@cd doseparse ; \
+	@if [ ! -e doseparseNoRpm ]; then $(LN) doseparse doseparseNoRpm ; fi
+	@cd doseparseNoRpm ; \
 		if [ ! -s doseparseNoRpm.mlpack ]; then \
 			$(LN) doseparse.mlpack doseparseNoRpm.mlpack ; \
 		fi ; \
diff -Naur a/configure b/configure
--- a/configure	2019-11-21 14:29:34.552324006 +0000
+++ b/configure	2019-11-21 14:29:34.616292009 +0000
@@ -3908,7 +3908,7 @@
  as_fn_error $? "Please install OCaml findlib module 'ocamlgraph'." "$LINENO" 5
 fi
 TMPVERSION=`$OCAMLFIND query -format %v ocamlgraph | sed 's/\.//g'`
-CONFIG_OCAMLGRAPH="-D 'OCAMLGRAPHVERSION $TMPVERSION'"
+CONFIG_OCAMLGRAPH="-D \"OCAMLGRAPHVERSION $TMPVERSION\""
 
 
 # Check whether --with-parmap was given.
@@ -5327,7 +5327,7 @@
 $as_echo_n "checking for a workable solution for ln -s... " >&6; }
   ln -s configure conftestLink
   if test "`cmd /c dir conftestLink 2>/dev/null | fgrep SYMLINK`" = ""; then :
-  LN=cp
+  LN="cp -a"
 else
   LN="ln -s"
 fi
diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2019-11-21 14:29:34.552324006 +0000
+++ b/configure.ac	2019-11-21 14:29:34.616292009 +0000
@@ -104,7 +104,7 @@
  AC_MSG_ERROR([Please install OCaml findlib module 'ocamlgraph'.])
 fi
 TMPVERSION=`$OCAMLFIND query -format %v ocamlgraph | sed 's/\.//g'`
-CONFIG_OCAMLGRAPH="-D 'OCAMLGRAPHVERSION $TMPVERSION'"
+CONFIG_OCAMLGRAPH="-D \"OCAMLGRAPHVERSION $TMPVERSION\""
 
 AC_ARG_WITH(parmap,
         [ --with-parmap ],
@@ -262,7 +262,7 @@
 AS_IF([test "${OCAML_OS_TYPE}" = "Win32"],[
   AC_MSG_CHECKING([for a workable solution for ln -s])
   ln -s configure conftestLink
-  AS_IF([test "`cmd /c dir conftestLink 2>/dev/null | fgrep SYMLINK`" = ""],[LN=cp],[LN="ln -s"])
+  AS_IF([test "`cmd /c dir conftestLink 2>/dev/null | fgrep SYMLINK`" = ""],[LN="cp -a"],[LN="ln -s"])
   AC_MSG_RESULT([$LN])
 ],[
   LN="ln -s"
