diff -Naur a/src/common/util.ml b/src/common/util.ml
--- a/src/common/util.ml	2021-07-22 07:14:19.000000000 +0000
+++ b/src/common/util.ml	2021-07-22 17:34:13.000000000 +0000
@@ -90,7 +90,7 @@
   let clean label =
     try
       let s = Filename.chop_extension (Filename.basename label) in
-      String.capitalize_ascii s
+      String.capitalize s
     with Invalid_argument _ -> label
 
   let create ?(enabled = false) label =
