diff -Naur a/src/context_flags.ml b/src/context_flags.ml
--- a/src/context_flags.ml	2024-08-21 13:03:37.000000000 +0000
+++ b/src/context_flags.ml	2024-09-05 14:57:29.242019552 +0000
@@ -23,7 +23,7 @@
   let flags =
     (if (Sys.win32 && Config.ccomp_type = "msvc")
      then ["\"/EHsc\""]
-     else ["-Wall -Wextra -Wno-unused-parameter -x c++"]) @
+     else ["-Wall -Wextra -Wno-unused-parameter -x c++ -std=gnu++11"]) @
     (ifc useGLPK ["-DUSEGLPK"]) @
     (ifc useCOIN ["-DUSECOIN"]) @
     (ifc useCLP  ["-DUSECLP"]) @
