diff -Naur a/common/criteria_lexer.mll b/common/criteria_lexer.mll
--- a/common/criteria_lexer.mll	2016-07-20 11:45:05.000000000 +0000
+++ b/common/criteria_lexer.mll	2019-11-21 14:29:33.856671969 +0000
@@ -18,7 +18,7 @@
     let c = Lexing.lexeme_char lexbuf 2 in (* the delimiter can be any character *)
     (* find the terminating delimiter *)
     let endpos =
-      try String.index_from lexbuf.lex_buffer (lexbuf.lex_start_pos + 3) c with
+      try Bytes.index_from lexbuf.lex_buffer (lexbuf.lex_start_pos + 3) c with
       |Invalid_argument _ ->
           raise (Format822.Syntax_error (
             Format822.error lexbuf "String too short"))
@@ -27,7 +27,7 @@
             Format822.error lexbuf (Printf.sprintf "cannot find: %c" c)))
     in
     let len = endpos - (lexbuf.lex_start_pos + 3) in
-    let s = String.sub lexbuf.lex_buffer (lexbuf.lex_start_pos + 3) len in
+    let s = Bytes.sub_string lexbuf.lex_buffer (lexbuf.lex_start_pos + 3) len in
     lexbuf.Lexing.lex_curr_pos <- lexbuf.Lexing.lex_start_pos + ((String.length s)+4);
     s
 
diff -Naur a/common/input.ml b/common/input.ml
--- a/common/input.ml	2016-07-20 11:45:05.000000000 +0000
+++ b/common/input.ml	2019-11-21 14:29:33.856671969 +0000
@@ -42,7 +42,7 @@
     with End_of_file -> raise IO.No_more_input
   in
   let read ch s pos len =
-    try Bz2.read ch s pos len 
+    try Bz2.read ch (Bytes.unsafe_to_string s) pos len 
     with End_of_file -> raise IO.No_more_input
   in
   IO.create_in
