diff -Naur a/src/dot_lexer.mll b/src/dot_lexer.mll
--- a/src/dot_lexer.mll	2020-10-02 13:47:57.000000000 +0000
+++ b/src/dot_lexer.mll	2022-11-30 10:28:54.843897528 +0000
@@ -18,6 +18,13 @@
 (* $Id:$ *)
 
 {
+#if OCAML_VERSION < (4, 3, 0)
+module String = struct
+  include String
+  let lowercase_ascii = lowercase
+end
+#endif
+
   open Dot_ast
   open Dot_parser
 
diff -Naur a/src/dune b/src/dune
--- a/src/dune	2020-10-02 13:47:57.000000000 +0000
+++ b/src/dune	2022-11-30 10:28:13.898840697 +0000
@@ -3,7 +3,10 @@
 (library
  (name graph)
  (public_name ocamlgraph)
+ (flags :standard -w -3)
  (libraries stdlib-shims)
+ (preprocess  (per_module
+                 ((action (run %{bin:cppo} -V OCAML:%{ocaml_version} %{input-file})) Dot_lexer)))
  (modules_without_implementation dot_ast sig sig_pack))
 
 (ocamlyacc dot_parser)
