diff -Naur a/src/cmdliner_docgen.ml b/src/cmdliner_docgen.ml
--- a/src/cmdliner_docgen.ml	2019-06-14 09:55:03.000000000 +0000
+++ b/src/cmdliner_docgen.ml	2019-11-21 14:29:33.193003934 +0000
@@ -146,7 +146,7 @@
   | Cmdliner_info.Val v ->
       match Lazy.force v with
       | "" -> strf "%s" (or_env ~value:false a)
-      | v -> strf "absent=%s%s" v (or_env ~value:true a)
+      | v -> strf "absent=%s%s" (esc v) (or_env ~value:true a)
   in
   let optvopt = match Cmdliner_info.arg_opt_kind a with
   | Cmdliner_info.Opt_vopt v -> strf "default=%s" v
