From 5d5d1d4a39830a9f852714ad6b14fec27a9ba0a5 Mon Sep 17 00:00:00 2001
From: Roberto Di Cosmo <github@dicosmo.org>
Date: Sun, 12 Jan 2014 22:54:38 +0100
Subject: [PATCH 01/10] Added portable invocation of mktemp for Linux and *BSD

---
 common/cudfSolver.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/cudfSolver.ml b/common/cudfSolver.ml
index 9b71147..8638f12 100644
--- a/common/cudfSolver.ml
+++ b/common/cudfSolver.ml
@@ -27,7 +27,7 @@ let check_fail file =
 (** see mktemp(1) for the syntax of [tmp_pattern] *)
 let mktmpdir tmp_pattern =
   let ic =
-    Unix.open_process_in (Printf.sprintf "mktemp --tmpdir -d %s" tmp_pattern) in
+    Unix.open_process_in (Printf.sprintf "(mktemp --tmpdir -d %s || mktemp -d -t %s) 2>/dev/null" tmp_pattern tmp_pattern) in
   let path = input_line ic in
   ignore (Unix.close_process_in ic);
   path
-- 
1.8.5.3

