diff -Naur a/src/context_flags.ml b/src/context_flags.ml
--- a/src/context_flags.ml	2021-04-16 13:58:45.000000000 +0000
+++ b/src/context_flags.ml	2022-01-27 20:47:25.606470269 +0000
@@ -37,6 +37,7 @@
 let clibs =
   let flags =
     (ifc (Config.ccomp_type = "cc") ["-lstdc++"]) @
+    (ifc (Config.system = "cygwin") ["-lgcc_s"]) @
     (ifc useCOIN ["-lCoinUtils"]) @
     (ifc useCLP  ["-lOsiClp"]) @
     (ifc useCBC  ["-lOsiCbc";"-lCbc"]) @
